packages/compiler/test/core/compiler-code-fixes/model-to-literal.codefix.test.ts

```diff
@@ -21,3 +21,39 @@ it("it change model expression to an object value", async () => {
       }
     `);
 });
+
+it("it recursively changes the model expression to the corresponding object value", async () => {
+  await expectCodeFixOnAst(
+    `
+      @example(┆{Bar : {Baz : "Hello"}})
+      model Foo { Bar : Bar; }
+      model Bar { Baz : string }
+    `,
+    (node) => {
+      strictEqual(node.kind, SyntaxKind.ModelExpression);
+      return createModelToObjectValueCodeFix(node);
+    },
+  ).toChangeTo(`
+      @example(#{Bar : #{Baz : "Hello"}})
+      model Foo { Bar : Bar; }
+      model Bar { Baz : string }
+    `);
+});
+
+it("it recursively changes the complex model expression to the corresponding object value or array value", async () => {
```

Comment:
It looks like it won't recursively fix items within a tuple.
Take this model for example:
```tsp
model Tuple {
  tuple: [ string, [ string ]] = ["foo", ["bar"]];
  obj: { tuple: [string, [string] ] } = { tuple: ["foo", ["bar"]] };
}
```
If I perform the codefix on the outermost object passed as the default to `obj`, then it works and I get `#{ tuple: #["foo", #["bar"]] };`.

However performing the codefix on the outermost array passed as the default to `tuple` does not recurse the elements in the tuple: `#["foo", ["bar"]];`

I noticed this when performing the codefix on a tuple within an object, so it seems to be an issue when the codefix is applied directly on a tuple vs an object.